scripts/pre_commit_linter.py

```diff
@@ -102,12 +102,47 @@
         'regexp': r"\b(ddescribe|fdescribe)\(",
         'message': "In tests, please use 'describe' instead of 'ddescribe'"
                    "or 'fdescribe'",
-        'excluded_files': ()
+        'excluded_files': (),
+        'excluded_dirs': ()
     },
     {
         'regexp': r"\b(iit|fit)\(",
         'message': "In tests, please use 'it' instead of 'iit' or 'fit'",
-        'excluded_files': ()
+        'excluded_files': (),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': r"templateUrl: \'",
+        'message': "The directives must be directly referenced.",
+        'excluded_files': (
+            'core/templates/dev/head/pages/exploration_editor/'
+            'editor_tab/StateResponses.js',
+            'core/templates/dev/head/pages/exploration_player/'
+            'FeedbackPopupDirective.js'
+            ),
+        'excluded_dirs': (
+            'extensions/')
+    }
+]
+
+BAD_PATTERNS_HTML_REGEXP = [
+    {
+        'regexp': r"text\/ng-template",
+        'message': "The directives must be directly referenced.",
+        'excluded_files': (
+            'core/templates/dev/head/pages/exploration_editor/'
+            'editor_tab/state_editor_responses.html',
+            'core/templates/dev/head/pages/exploration_player/'
```

Comment:
Why exclude this one?

Reply:
It uses popover/feedback directive. 
Oh, now I realized that it doesn't have `text/ng-template` so need not be excluded explicitly.

Removed from the list. thanks.

Reply:
Not removed?

Reply:
This file has `text/ng-template` so it has to be excluded explicitly. Others have been removed from list (eg `/pages/exploration_player/conversation_skin_directive.html` is removed even though it uses popover/feedback`)